File: Sorting/Insertion-Sort/InsertionSort.py
"""
Insertion sort in python
Space complexity - O(1)
Time complexity - O(n^2)
"""


def insertion_sort(arr):
    for i in range(1, len(arr)):
        j = i
        while j > 0 and arr[j] < arr[j-1]:
            arr[j], arr[j-1] = arr[j-1], arr[j]
            j -= 1
    return arr


if __name__ == "__main__":
    print(insertion_sort([4, 6, 1, 7, 3, 19, 11]))

File: Sorting/Insertion-Sort/InsertionSort.ts
// Insertion sort using TypeScript
// space complexity - O(1)
// Time Complexity - O(n^2)

export function insertionSort(arr:number[]):number[]{
    for(let i = 0; i< arr.length; i++){
        let j = i;
        while (j > 0 && arr[j] < arr[j-1]){
            //swap the values
            [arr[j],arr[j-1]]  = [arr[j-1],arr[j]];
            j -= 1;
        }
    }
    return arr;
}

console.log(insertionSort([2,6,3,4,1,11,17,10]));